pkg/ruler/manager.go

```diff
@@ -141,33 +137,50 @@ func (r *DefaultMultiTenantManager) syncRulesToManager(ctx context.Context, user
 		return
 	}
 
+	manager, done := r.getOrCreateManager(ctx, user, update)
+	if done {
+		return
+	}
+	err = manager.Update(r.cfg.EvaluationInterval, files, nil, r.cfg.ExternalURL.String(), nil)
+	if err != nil {
+		r.lastReloadSuccessful.WithLabelValues(user).Set(0)
+		level.Error(r.logger).Log("msg", "unable to update rule manager", "user", user, "err", err)
+		return
+	}
+
+	r.lastReloadSuccessful.WithLabelValues(user).Set(1)
+	r.lastReloadSuccessfulTimestamp.WithLabelValues(user).SetToCurrentTime()
+}
+
+// getOrCreateManager retrieves the user manager. if it doesn't exist, it will create and start it first
+func (r *DefaultMultiTenantManager) getOrCreateManager(ctx context.Context, user string, update bool) (RulesManager, bool) {
```

Comment:
No need to return the `done` boolean. Let's move all update logic to the caller. What you should do instead is returning an `error` on error.